src/compiler/crystal/compiler.cr

```diff
@@ -421,6 +421,8 @@ module Crystal
       output_filename = File.expand_path(output_filename)
 
       @progress_tracker.stage("Codegen (linking)") do
+        # Ensures that the linking step is reproducible from verbose output.
+        print_command(%(cd "${@}"), output_dir) if verbose?
```

Comment:
This doesn't work on Windows, and because `cd` is a built-in command there, `Process.quote` is inapplicable here. The working directory therefore must be displayed separately without any reference to `cd`.